_posts/blog/2016-05-16-animations_touch.md

```diff
@@ -0,0 +1,103 @@
+
+Implementing a view translation can be easily done programmatically.
+{% highlight java %}
+view.animate().translationYBy(animateByPx);
+{% endhighlight %}
+Where the ``animateByPx`` is the amount of pixels the view is translated by on the Y coordinate.
+
+##Handling Touch Events
+
+The main difference between view and property animations lies in whether the ``View`` is modified or not. With view animations, the ``View`` gets drawn in another position, but the actual ``View`` does not move. So, this means that the view will not react to touch events on the area where it is drawn, but rather in the area where it was before the animation started.
+With property animations, the object actually gets moved. Therefore it will react to touch events on its new location on the screen.
+
+To see better how the touch area changes, we have implemented a selector for our ``View``’s background, allowing us to set two different colors depending on the pressed state of the view: pink for pressed and blue for not pressed.
+
+With the view animations implementation, after the translation, when touching the area that is above the original location of the ``View``, the ``View`` does not react and the touch events are caught by the area outside it.
+<picture>
+	<img src="/images/animations_touch/animations_touch_above.gif" alt="image">
+  <figcaption>Touch events on the new position of the view, after a view animation was applied.</figcaption>
+</picture>
+
+When touching the area containing the initial position of the ``View``, we see that the 25% of the ``View`` - that is still drawn over the original position - reacts to our touch events.
+<picture>
+</picture>
+<center>
+<picture class="half">
+  <img src="/images/animations_touch/animations_touch_initial_position.gif" alt="image">
+	<figcaption>Touch events on the initial position of the view, after a view animation was applied.</figcaption>
+</picture>
+</center>
+
+With the property animations implementation we see that the ``View`` reacts to touch events exactly as expected, no matter where the touch events happen.
+<center>
+<picture class="half">
+<img src="/images/animations_touch/animations_touch_property.gif" alt="image">
+	<figcaption>Touch events with property animations.</figcaption>
+</picture>
+</center>
+
+##Conclusion
+
+Both view and property animations are easy to implement. Use the one that’s more convenient for you, but when needing to implement touch events on a ``View`` that was already animated, you should use **property animations**.
+
+Check out the implementation of view and property translation animations in this <a href="https://github.com/florina-muntenescu/Playground">GitHub repo</a>.
```

Comment:
maybe we could think about adding a playground repo to the upday github account for stuff like this - but this is up to discussion